pkg/ucp/frontend/controller/credentials/aws/createorupdateawscredential.go

```diff
@@ -84,7 +84,9 @@ func (c *CreateOrUpdateAWSCredential) Run(ctx context.Context, w http.ResponseWr
 	}
 
 	// Do not save the secret in metadata store.
-	newResource.Properties.AWSCredential.SecretAccessKey = ""
+	if newResource.Properties.AWSCredential.Kind == datamodel.AWSAccessKeyCredentialKind {
+		newResource.Properties.AWSCredential.AccessKeyCredential.SecretAccessKey = ""
+	}
```

Comment:
Do we have an existing unit test for this check (to make sure we don't accidentally start storing secrets)?

Reply:
the tests in /radius/pkg/ucp/frontend/controller/credentials/aws/createorupdateawscredential_test.go use  func getAwsResponse() armrpc_rest.Response as the expected response, which makes sure the structure does not have the secret.